File: core/src/main/java/vn/loitp/views/ldebugview/LDebug.java
package vn.loitp.views.ldebugview;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.provider.Settings;

import vn.loitp.utils.util.ServiceUtils;

/**
 * Created by LENOVO on 7/5/2018.
 */

public class LDebug {
    public final static int CODE = 1993;

    public static void init(Activity activity) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.canDrawOverlays(activity)) {
            Intent intent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:" + activity.getPackageName()));
            activity.startActivityForResult(intent, CODE);
        } else {
            ServiceUtils.startService(LDebugViewService.class);
        }
    }

    public static void checkPermission(Activity activity, int requestCode, int resultCode) {
        if (requestCode == CODE) {
            init(activity);
        }
    }

    public static void stop() {
        ServiceUtils.stopService(LDebugViewService.class.getName());
    }

    public static void log(String log) {
        ComunicateDebug.postFromActivity(new ComunicateDebug.MsgFromActivity(ComunicateDebug.MsgFromActivity.TYPE_D, log, null));
    }

    public static void log(int type, String log) {
        ComunicateDebug.postFromActivity(new ComunicateDebug.MsgFromActivity(type, log, null));
    }

    public static void log(Object o) {
        ComunicateDebug.postFromActivity(new ComunicateDebug.MsgFromActivity(ComunicateDebug.MsgFromActivity.TYPE_D, "", o));
    }

    public static void log(int type, Object o) {
        ComunicateDebug.postFromActivity(new ComunicateDebug.MsgFromActivity(type, "", o));
    }
}

File: core/src/main/java/com/loitp/views/lDebugView/LDebug.kt
package com.loitp.views.lDebugView

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.provider.Settings
import com.loitp.core.ext.startService
import com.loitp.core.ext.stopService
import com.loitp.views.lDebugView.LComunicateDebug.postFromActivity

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
object LDebug {

    private fun isNeedPermission(activity: Activity): Boolean {
        return !Settings.canDrawOverlays(activity)
    }

    @JvmStatic
    fun init(
        activity: Activity,
        data: ((Intent) -> Unit)? = null,
    ) {
        if (isNeedPermission(activity)) {
            val intent = Intent(
                Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
                Uri.parse("package:" + activity.packageName)
            )
            data?.invoke(intent)
        } else {
            start(activity)
        }
    }

    @JvmStatic
    fun start(context: Context) {
        context.startService(LDebugViewService::class.java.name)
    }

    @JvmStatic
    fun stop(context: Context) {
        context.stopService(LDebugViewService::class.java.name)
    }

    fun log(log: String?) {
        postFromActivity(
            msg = LComunicateDebug.MsgFromActivity(
                type = LComunicateDebug.MsgFromActivity.TYPE_D,
                msg = log,
                any = null
            )
        )
    }

    fun log(
        type: Int,
        log: String?
    ) {
        postFromActivity(
            msg = LComunicateDebug.MsgFromActivity(
                type = type,
                msg = log,
                any = null
            )
        )
    }

    fun log(o: Any?) {
        postFromActivity(
            msg = LComunicateDebug.MsgFromActivity(
                type = LComunicateDebug.MsgFromActivity.TYPE_D,
                msg = "",
                any = o
            )
        )
    }

    fun log(
        type: Int,
        o: Any?
    ) {
        postFromActivity(
            msg = LComunicateDebug.MsgFromActivity(
                type = type,
                msg = "",
                any = o
            )
        )
    }
}
